commcare_connect/opportunity/visit_import.py

```diff
@@ -271,12 +271,18 @@ def _bulk_update_payments(opportunity: Opportunity, imported_data: Dataset) -> P
             payment = Payment.objects.create(opportunity_access=access, amount=amount, amount_usd=amount_usd)
             seen_users.add(username)
             payment_ids.append(payment.pk)
-            update_work_payment_date(access)
+        process_work_payment_dates(users)
```

Comment:
Moving this function outside the transaction block would be a better solution to this problem. Reducing the amount of work done in a transaction over multiple tables will reduce the chances of a deadlock. Also `update_work_payment_date` calls a [`bulk_update/bulk_create`](https://github.com/django/django/blob/main/django/db/models/query.py#L911) function that also uses a transaction and would fail if it tries to access an object from the transaction before.